geode-core/src/main/java/org/apache/geode/DataSerializer.java

```diff
@@ -460,22 +460,13 @@ public static void writeInetAddress(InetAddress address, DataOutput out) throws
   public static InetAddress readInetAddress(DataInput in) throws IOException {
 
     InternalDataSerializer.checkIn(in);
+    InetAddress address = StaticSerialization.readInetAddress(in);
 
-    byte[] address = readByteArray(in);
-    if (address == null) {
-      return null;
-    }
-
-    try {
-      InetAddress addr = InetAddress.getByAddress(address);
-      if (logger.isTraceEnabled(LogMarker.SERIALIZER_VERBOSE)) {
-        logger.trace(LogMarker.SERIALIZER_VERBOSE, "Read InetAddress {}", addr);
-      }
-      return addr;
-    } catch (UnknownHostException ex) {
-      throw new IOException("While reading an InetAddress", ex);
```

Comment:
Was this exception handling superfluous?

Reply:
yes - not needed. We already handle this exception further up the call stack